
# HG changeset patch
# User Michael Elkins <me@mutt.org>
# Date 1285942464 25200
# Node ID 1a4c43138685b628dedb14932934ff392104ac72
# Parent  c10f8863e979e6f7fa61649128e3b659b4cd8827
fix segfault when $message_cachedir is set and opening a POP3 mailbox.

fixes #3457

diff -r c10f8863e979 -r 1a4c43138685 muttlib.c
--- a/muttlib.c	Thu Sep 30 18:11:35 2010 -0700
+++ b/muttlib.c	Fri Oct 01 07:14:24 2010 -0700
@@ -1960,6 +1960,7 @@
 {
   char *p = safe_strdup (src);
   int rc = mutt_convert_string (&p, Charset, "utf-8", 0);
-  strfcpy (dest, rc == 0 ? p : src, dlen);
+  /* `src' may be NULL, such as when called from the pop3 driver. */
+  strfcpy (dest, (rc == 0) ? NONULL(p) : NONULL(src), dlen);
   FREE (&p);
 }

